src/libstd/path.rs

```diff
@@ -1148,32 +1171,34 @@ impl Path {
     /// use std::path::Path;
     ///
     /// let path = Path::new("/foo/bar");
-    /// let foo = path.parent().unwrap();
+    /// let foo = path.without_file().unwrap();
     /// assert!(foo == Path::new("/foo"));
-    /// let root = foo.parent().unwrap();
+    /// let root = foo.without_file().unwrap();
     /// assert!(root == Path::new("/"));
-    /// assert!(root.parent() == None);
+    /// assert!(root.without_file() == None);
     /// ```
-    pub fn parent(&self) -> Option<&Path> {
+    #[stable(feature = "rust1", since = "1.0.0")]
+    pub fn without_file(&self) -> Option<&Path> {
```

Comment:
We talked about this a bit  on IRC, but I'm a bit worried about bucking the trend here. I've been unable to find any precedent for the parent of "foo/.." being "foo/.." (all seem to chop off the "/.."). I do agree that this is "morally correct" but it seems odd to differ.

I've also been somewhat worried about this returning an `Option` in the past, but I'm mostly ok with that doing so given the understanding that the root path's parent is `None`.

I would personally stick to `parent` here and keep basically same semantics here, or at least that's my vote!